src/mlpack/methods/elm/elm.cpp

```diff
@@ -0,0 +1,201 @@
+	    }
+	    break;
+
+    case 1 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = sin(param(i,j)+bias(j));
+	     }
+	    }
+	    break;
+
+    case 2 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = ((param(i,j)+bias(j)) > 0)? 1 : 0;
+	     }
+	    }
+	    break;
+
+    case 3 :for(size_t i=0;i<H.n_rows;++i)
+	    {
+	     for(size_t j=0; j<H.n_cols;++j)
+	     {
+	      H(i,j) = ((param(i,j)+bias(j)) <= 1)&&((param(i,j)+bias(j)) >= -1) ? (1-abs(param(i,j)+bias(j))) : 0.0;
+	     }
+	    }
+	    break;
+
+    case 4 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = exp(- ((param(i,j)+bias(j))*(param(i,j)+bias(j))));
+             }
+	    }
+	    break;
+
+    default : Log::Fatal << "Please select a suitable activation function to proceed:" << std::endl;
+  }
+
+  mat H_inv = pinv(H); // Moore-Penrose pseudo-inverse of matrix H
+  beta = H_inv * responses; //Calculate output weights 
+  mat trainingOutput = H * beta;  // Calculate training accuracy
+  vec temp = responses - trainingOutput;
+  double trainError = stddev(temp); //calculate training error
+  Log::Info << "Train RMSE :" << trainError <<std::endl;
+}
+
+/*
+ Predict ELM
+ Data set points and predictions; 
+ Activation function
+		0 - Sigmoid Function
+		1 - Sine Function
+		2 - Hardlim Function
+		3 - Triangular Bias Function
+		4 - Radial Basis Function
+*/
+
+void ELM::Predict(const arma::mat& points,
+                  const arma::mat& predictions)
+{
+  mat param = points*weight.t(); 
+  mat H = zeros(N,Nh);
+	 	
+  switch(act)
+  {
+    case 0 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+             {
+	      H(i,j) = 1.0 / (1.0 + exp(- (param(i,j)+bias(j))));
+	     }
+	    }
+	    break;
+
+    case 1 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = sin(param(i,j)+bias(j));
+	     }
+	    }
+	    break;
+
+    case 2 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = ((param(i,j)+bias(j)) > 0)? 1 : 0;
+	     }
+	    }
+	    break;
+
+    case 3 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+	     {
+	      H(i,j) = ((param(i,j)+bias(j)) <= 1)&&((param(i,j)+bias(j)) >= -1) ? (1-abs(param(i,j)+bias(j))) : 0.0;
+	     }
+	    }
+	    break;
+
+    case 4 :for(size_t i=0; i<H.n_rows; ++i)
+	    {
+	     for(size_t j=0; j<H.n_cols; ++j)
+  	     {
+	      H(i,j) = exp(- ((param(i,j)+bias(j) - alpha)*(param(i,j)+bias(j) - alpha))/lambda);
+	     }
+	    }
+	    break;
+
+    default : Log::Fatal << "Please select a suitable activation function to proceed:" << std::endl;
+  }
+
+  mat predictions = H * beta; // predict the output of ELM
+  arma::mat Elm_output = arma::randu<arma::mat>(N);
+  Elm_output = predictions;
```

Comment:
Please do not save the output to the csv after predict but let the user have the output results directly.
I suggest you change the api to 

`void ELM::Predict(const arma::mat& points, arma::mat& predictions);`